### ISSUE TITLE
Build error in Django web project with new project from existing

### ISSUE THREAD

**Issue by zhouwangyang** [microsoft/PTVS#6045]
**Steps to Reproduce**
1.Create Django web project named ProjectA
2.Add virtual environment with python 3.7 as current environment
3.File, New Project, Python, From Existing Python Code
4.Point to the generated project ProjectA, and make sure to select "Customize Project Type: Django Web Project" on the last page
5.Build solution

**Expected behavior**
Solution should be build successfully.

**Additional context and screenshots**
Build FAILED. error in error list.
`------ Rebuild All started: Project: DjangoWebProject1, Configuration: Debug Any CPU ------
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\MSBuild\Microsoft\VisualStudio\v16.0\Python Tools\Microsoft.PythonTools.Django.targets(73,5): error MSB4132: The "RunPythonCommand" task returned false but did not log an error.
Done building project "DjangoWebProject1.pyproj" -- FAILED.

Build FAILED.
========== Rebuild All: 0 succeeded, 1 failed, 0 skipped ==========`
![image](https://user-images.githubusercontent.com/39871640/76758120-026a6700-67c4-11ea-8250-7d429c701f4f.png)

**Environment**
![image](https://user-images.githubusercontent.com/39871640/76758191-3180d880-67c4-11ea-8cfa-bf3cbaa992e3.png)

**Comment by zhouwangyang** [comment]
can be reproduced in the latest build:
Build of Visual Studio: 16.6.0 Preview 3.0 [29923.199.master]
Build of Python Package: 16.5.20042.1

**Comment by AdamYoblick** [comment]
I was able to repro this as well.

@bschnurr This seems to be an issue in the way Visual Studio creates a new project Django project from existing code. Building/running a newly created Django project seems to work just fine.

![image](https://user-images.githubusercontent.com/29436557/102152550-dfe0a980-3e3a-11eb-9b40-603a0b1503de.png)

I'm not sure what the best way to debug this is, since this is a build error and setting a breakpoint in the PTVS source won't work. Should I just be diffing the pyproj files or reach out to Hugues?

**Comment by bschnurr** [comment]
Ya I think your best bet to compare proj files

**Comment by bschnurr** [comment]
maybe see if a non-django project "from exiting python code" works too

**Comment by AdamYoblick** [comment]
Ok I'm still not sure what the cause of the bug is, but it seems that creating a new project from existing code does not build correctly unless the project is saved right next to the original project. Maybe some relative path issues, I need to investigate further.

Also, the project name matters. Since there is python code that has hardcoded `ProjectA.settings`, if your project is named something else, it will not work. I verified this by soing the following:

1. Creating a new project from existing code (ProjectA1)
2. Saving that pyproj file right next to ProjectA.pyproj
3. Then deleting the original ProjectA.pyproj
4. And renaming ProjectA1.pyproj to ProjectA.pyproj.

So this is a workaround.

Another workaround is to just create a new django web project from scratch, which definitely does build, then copy existing files into that new project.

I will continue investigating this to see if I can find the root cause.

**Comment by AdamYoblick** [comment]
@bschnurr "from existing python code" works just fine with a python console application, for example. The only main difference in the pyproj is the `<ProjectHome>` which points at the existing code specified when creating the project.

![image](https://user-images.githubusercontent.com/29436557/102268982-9dbe7300-3ee1-11eb-8153-02425d20e910.png)

![image](https://user-images.githubusercontent.com/29436557/102268943-913a1a80-3ee1-11eb-8bcd-2b9bcaaebe73.png)

**Comment by AdamYoblick** [comment]
I was able to get around the first build error by doing two things:

1. Add the following property to your new (ProjectB) pyproj file:

`<DjangoSettingsModule>ProjectA.settings</DjangoSettingsModule>`

Where `ProjectA` should be the name of the existing project that you used.

The default value for this property is defined in `C:\Program Files (x86)\Microsoft Visual Studio\2019\PubPreview\MSBuild\Microsoft\VisualStudio\v16.0\Python Tools\Microsoft.PythonTools.Django.targets` as the following:

`<DjangoSettingsModule Condition="'$(DjangoSettingsModule)' == ''">$(MSBuildProjectName).settings</DjangoSettingsModule>`

That's why the build is trying to import ProjectB.settings, because is uses the new project name to build the file name. You have to force it to use the old project name since that's the real file that exists.

2. Add the "existing" project folder as a search path of the new project:

![image](https://user-images.githubusercontent.com/29436557/102276071-ee3ace00-3eeb-11eb-95b6-7d31751f410f.png)

This is needed so the `import ProjectA.settings` statement works properly.

Here's the diagnostic output of the command that was failing before, which is passing now:

```
Target "DetectStaticRootPath" in file "C:\Program Files (x86)\Microsoft Visual Studio\2019\PubPreview\MSBuild\Microsoft\VisualStudio\v16.0\Python Tools\Microsoft.PythonTools.Django.targets" from project "C:\Users\advolker\Source\Repos\ProjectB\ProjectB.pyproj" (target "DjangoCollectStaticFiles" depends on it):
	Initializing task factory "RunPythonCommandFactory" from assembly "C:\Program Files (x86)\Microsoft Visual Studio\2019\PubPreview\Common7\IDE\Extensions\Microsoft\Python\Core\Microsoft.PythonTools.BuildTasks.dll".
	Using "RunPythonCommand" task from the task factory "RunPythonCommandFactory".
	Task "RunPythonCommand"
		Task Parameter:Environment=DJANGO_SETTINGS_MODULE=ProjectA.settings
		Task Parameter:Target=import ProjectA.settings as settings; print(getattr(settings, 'STATIC_ROOT', '') or '')
		Task Parameter:ConsoleToMSBuild=True
		Task Parameter:ExecuteIn=none
		Task Parameter:TargetType=code
		Task Parameter:WorkingDirectory=.
		Output Property: DjangoStaticRootSetting=C:/Users/advolker/Source/Repos/ProjectA/ProjectA/static
	Done executing task "RunPythonCommand".
```

There is a subsequent build error that I'm looking into now.

**Comment by AdamYoblick** [comment]
The next error I see is:

```
Done building target "SetStartupPathOrManagePy" in project "ProjectB.pyproj".
Target "_DjangoCollectStaticFiles" in file "C:\Program Files (x86)\Microsoft Visual Studio\2019\PubPreview\MSBuild\Microsoft\VisualStudio\v16.0\Python Tools\Microsoft.PythonTools.Django.targets" from project "C:\Users\advolker\Source\Repos\ProjectB\ProjectB.pyproj" (target "DjangoCollectStaticFiles" depends on it):
	Task "RunPythonCommand"
		Task Parameter:Environment=DJANGO_SETTINGS_MODULE=ProjectA.settings
		Task Parameter:Arguments=collectstatic --noinput
		Task Parameter:Target=C:\Users\advolker\Source\Repos\ProjectA\manage.py
		Task Parameter:ExecuteIn=none
		Task Parameter:TargetType=script
		Task Parameter:WorkingDirectory=.
		C:\Users\advolker\AppData\Local\Programs\Python\Python37\python.exe: can't open file 'C:\Users\advolker\Source\Repos\ProjectA\manage.py': [Errno 2] No such file or directory
C:\Program Files (x86)\Microsoft Visual Studio\2019\PubPreview\MSBuild\Microsoft\VisualStudio\v16.0\Python Tools\Microsoft.PythonTools.Django.targets(99,5): error MSB4181: The "RunPythonCommand" task returned false but did not log an error.
	Done executing task "RunPythonCommand" -- FAILED.
```

Looking at `C:\Program Files (x86)\Microsoft Visual Studio\2019\PubPreview\MSBuild\Microsoft\VisualStudio\v16.0\Python Tools\Microsoft.PythonTools.Django.targets`, here's the code that runs:

```
  <Target Name="_DjangoCollectStaticFiles"
          DependsOnTargets="SetStartupPathOrManagePy"
          Condition="'$(DjangoStaticRootSetting)' != ''">
    <RunPythonCommand Target="$(StartupPathOrManagePy)"
                      TargetType="script"
                      Arguments="collectstatic --noinput"
                      ExecuteIn="none"
                      WorkingDirectory="$(WorkingDirectory)"
                      Environment="DJANGO_SETTINGS_MODULE=$(DjangoSettingsModule)" />
```

This depends on `$(StartupPathOrManagePy)`, which is defined on line 133 of the same file:

```
    <StartupPathOrManagePy>$(QualifiedProjectHome)manage.py</StartupPathOrManagePy>
    <StartupPathOrManagePy Condition="Exists($(StartupPath))">$(StartupPath)</StartupPathOrManagePy>
```

`QualifiedProjectHome` is defined on line 134 of `C:\Program Files (x86)\Microsoft Visual Studio\2019\PubPreview\MSBuild\Microsoft\VisualStudio\v16.0\Python Tools\Microsoft.PythonTools.targets`, but it doesn't really matter. The second line overwrites it if startup path exists. So right click on `ProjectA\ProjectA\manage.py` and set it as the startup file, and the build succeeds.

After that, running with F5 launches the website as expected. I'll summarize the complete solution in the next comment.

**Comment by AdamYoblick** [comment]
tl;dr workaround:

1. Add the following property to your new (ProjectB) pyproj file: `<DjangoSettingsModule>ProjectA.settings</DjangoSettingsModule>`, where ProjectA should be the name of the existing project that you used.
2. Add the "existing" project folder as a search path of the new project:
![image](https://user-images.githubusercontent.com/29436557/102276071-ee3ace00-3eeb-11eb-95b6-7d31751f410f.png)
3. Right click on ProjectA\ProjectA\manage.py and set it as the startup file

After doing these three steps, the error no longer repros, and I can build and run as expected, so I'm going to close this out.
